util/tele/composite_logger.go

```diff
@@ -35,7 +35,11 @@ func (c *compositeLogger) Enabled() bool {
 
 func (c *compositeLogger) iter(fn func(l logr.Logger)) {
 	for _, l := range c.loggers {
-		fn(l)
+		if callDepthLogger, ok := l.(logr.CallDepthLogger); ok {
```

Comment:
You literally have to read to the end of this giant PR to find the one line of code that makes you go, "Hey, what's that do?"
You explained it to me, but maybe we should make a code comment here explaining why we try to default to the call-depth logger, 3 deep.

Reply:
related to https://github.com/kubernetes-sigs/cluster-api-provider-azure/issues/1827

Reply:
I've added a comment to explain why we are using the `CallDepthLogger` interface, and what it does.